pkg/pss/pss_test.go

```diff
@@ -0,0 +1,223 @@
+	"github.com/ethersphere/bee/pkg/storage"
+	"github.com/ethersphere/bee/pkg/storage/mock"
+	"github.com/ethersphere/bee/pkg/swarm"
+	"github.com/ethersphere/bee/pkg/tags"
+	"github.com/ethersphere/bee/pkg/trojan"
+)
+
+// TestTrojanChunkRetrieval creates a trojan chunk
+// mocks the localstore
+// calls pss.Send method and verifies it's properly stored
+func TestTrojanChunkRetrieval(t *testing.T) {
+	var err error
+	ctx := context.TODO()
+	testTags := tags.NewTags()
+
+	localStore := mock.NewTagsStorer(testTags)
+	pss := pss.NewPss(localStore, testTags)
+
+	target := trojan.Target([]byte{1}) // arbitrary test target
+	targets := trojan.Targets([]trojan.Target{target})
+	payload := []byte("RECOVERY CHUNK")
+	topic := trojan.NewTopic("RECOVERY TOPIC")
+
+	// call Send to store trojan chunk in localstore
+	if _, err = pss.Send(ctx, targets, topic, payload); err != nil {
+		t.Fatal(err)
+	}
+
+	var chunkAddress swarm.Address
+	// this code iterates over the localstore
+	// this will get the chunk that was stored in pss.Send
+	for po := uint8(0); po <= swarm.MaxPO; po++ {
+		last, err := localStore.LastPullSubscriptionBinID(po)
+		if err != nil {
+			t.Fatal(err)
+		}
+		if last == 0 {
+			continue
+		}
+		// iter for chunk in localstore
+		ch, _, _ := localStore.SubscribePull(context.Background(), po, 0, last)
+		for c := range ch {
+			chunkAddress = c.Address
+			break
+		}
+	}
+
+	// verify store, that trojan chunk has been stored correctly
+	var storedChunk swarm.Chunk
+	if storedChunk, err = localStore.Get(ctx, storage.ModeGetRequest, chunkAddress); err != nil {
+		t.Fatal(err)
+	}
+
+	// create a stored chunk artificially
+	m, err := trojan.NewMessage(topic, payload)
+	if err != nil {
+		t.Fatal(err)
+	}
+	var tc swarm.Chunk
+	tc, err = m.Wrap(targets)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	tag, err := tags.NewTags().Create("pss-chunks-tag", 1, false)
+	if err != nil {
+		t.Fatal(err)
+	}
+	storedChunk = tc.WithTagID(tag.Uid)
+
+	if !reflect.DeepEqual(tc, storedChunk) {
+		t.Fatalf("store chunk does not match sent chunk")
+	}
+}
+
+// TestPssMonitor creates a trojan chunk
+// mocks the localstore
+// calls pss.Send method
+// updates the tag state (Stored/Sent/Synced)
+// waits for the monitor to notify the changed state
+func TestPssMonitor(t *testing.T) {
+	var err error
+	ctx := context.TODO()
+	testTags := tags.NewTags()
+
+	localStore := mock.NewTagsStorer(testTags)
+
+	target := trojan.Target([]byte{1}) // arbitrary test target
+	targets := trojan.Targets([]trojan.Target{target})
+	payload := []byte("RECOVERY CHUNK")
+	topic := trojan.NewTopic("RECOVERY TOPIC")
+
+	var monitor *pss.Monitor
+
+	pss := pss.NewPss(localStore, testTags)
+
+	// call Send to store trojan chunk in localstore
+	if monitor, err = pss.Send(ctx, targets, topic, payload); err != nil {
+		t.Fatal(err)
+	}
+
+	storeTags := testTags.All()
+	if len(storeTags) != 1 {
+		t.Fatalf("expected %d tags got %d", 1, len(storeTags))
+	}
+
+	timeout := 1 * time.Second
```

Comment:
may I suggest `context.WithDeadline` for this?

Reply:
replaced